### ISSUE TITLE
instrument.js hangs sometimes in node v5

### ISSUE THREAD

**Issue by esbena** [Samsung/jalangi2#71]
Possibly related: #58 

instrument.js hangs in node v5

To reproduce:

```bash
$ node --version
v5.0.0
$ mkdir test
$ cd test
$ echo '{}' > package.json
$ npm i jalangi2
$ npm i minimist
$ node node_modules/jalangi2/src/js/commands/instrument.js --outputDir out node_modules/minimist
^C # it hangs..
$ rm -rf node_modules/minimist/test
$ node node_modules/jalangi2/src/js/commands/instrument.js --outputDir out node_modules/minimist
instrumenting out/minimist2/index_orig_.js
instrumenting out/minimist2/example/parse_orig_.js
done!
```

It seems that the call to ncp on line 454 of instrument.js never calls the callback. 
That can be shown with the following change:
```javascript
ncp(inputDir, copyDir, {transform: transform}, function(){console.log('ncp done')});
```

**Comment by msridhar** [comment]
Does it work with v4?  In the short-term probably downgrading node is your best bet.  I think the right fix is to get rid of our dependence on ncp entirely.  One simple solution is to do the following instead:

1. First just do a simple `cp -r` to the output directory (there must be a robust library for this).
2. Go through the output directory and replace files with their instrumented versions.

I think that will be a lot more robust and conceptually simple than what we do now.

**Comment by esbena** [comment]
https://www.npmjs.com/package/graceful-ncp is a drop-in replacement for ncp, just use `require('graceful-ncp')` instead. 
The issue is fixed if instrument.js is modified to use that instead of ncp.

I will make a pull-request when I have tested this fix for some time.

**Comment by msridhar** [comment]
Fixed by 619d5d235c2f1648c0c9f91cc1ea2f0591eb21df
